src/backend/cuda/compile_module.cpp

```diff
@@ -316,40 +316,46 @@ Module compileModule(const string &moduleKey, const vector<string> &sources,
                                  getKernelCacheFilename(device, moduleKey);
         const string tempFile =
             cacheDirectory + AF_PATH_SEPARATOR + makeTempFilename();
-
-        // compute CUBIN hash
-        const size_t cubinHash = deterministicHash(cubin, cubinSize);
-
-        // write module hash(everything: names, code & options) and CUBIN data
-        ofstream out(tempFile, std::ios::binary);
-        if (!sourceIsJIT) {
-            size_t mangledNamesListSize = retVal.map().size();
-            out.write(reinterpret_cast<const char *>(&mangledNamesListSize),
-                      sizeof(mangledNamesListSize));
-            for (auto &iter : retVal.map()) {
-                size_t kySize   = iter.first.size();
-                size_t vlSize   = iter.second.size();
-                const char *key = iter.first.c_str();
-                const char *val = iter.second.c_str();
-                out.write(reinterpret_cast<const char *>(&kySize),
-                          sizeof(kySize));
-                out.write(key, iter.first.size());
-                out.write(reinterpret_cast<const char *>(&vlSize),
-                          sizeof(vlSize));
-                out.write(val, iter.second.size());
+        try {
+            // write module hash(everything: names, code & options) and CUBIN
+            // data
+            ofstream out(tempFile, std::ios::binary);
```

Comment:
I was looking at the files generated by our code and these aren't cubin files because we add a bunch of information in the header. We should either have a separate extension or put the information in another file. Making the extension cubin is misleading.

Reply:
Not in this PR of course.

Reply:
Just an .bin extension should clear any confusion. It will be just a binary file saved to disk by arrayfire, the name has enough information o. What the binary is about already.